### DISCUSSION TITLE
How to update a score?

### DISCUSSION THREAD

**Discussion by leichx** [langfuse/langfuse#4178] [Support] [answered]
### Describe the feature or potential improvement

Hello, so we are using langfuse for our prompt management and I am playing with scoring and realized that is not possible to update an score, I have tried using observationId and traceId but it is always adding a new record and not updating the existing one, so the request is to create updateScore() or allow to update the existing score referencing to it with the observationId.

Thanks.

**Comment by marcklingen** [comment, marked as answer]
You can upsert scores based on their `id`.

You can either add an id yourself when creating a score and then using the same id again to update it, or fetch the score response to get access to the automatically created id to upsert it in a subsequent call via the sdk.

**Reply by leichx** [reply]
Thanks Marck, also is there a way to directly get the prompt with that id? Instead ot fetch all the trace data?

**Reply by marcklingen** [reply]
If the score relates to a single generation within a trace (score includes traceId and optionally also an observationId), you can obtain the prompt using its observation ID. This page may assist you with the data model: https://langfuse.com/docs/tracing
